Unable to write queue file, EACCES

### system info

Haraka | Haraka.js — Version: 2.8.24
 --- | :--- 
Node | v10.16.0
OS | Linux ba33c2ee7ccf 4.15.0-1060-aws #62-Ubuntu SMP Tue Feb 11 21:23:22 UTC 2020 x86_64 GNU/Linux
openssl | OpenSSL 1.1.0j  20 Nov 2018

### Expected behavior

Sending outbound mail should work.

### Observed behavior

Queueing fails:

~~~
[ERROR] [-] [core] [outbound] Unable to write queue file (1584147605715_1584147605715_0_19_kSiOk1_2_93af1f9d0b49): Error: EACCES: permission denied, open '/app/queue/__tmp__.1584147605715_1584147605715_0_19_kSiOk1_2_93af1f9d0b49'
[ERROR] [7D41E177-89D3-4D7D-8AA4-7999E1945850.1] [outbound] Queueing failed
[ERROR] [7D41E177-89D3-4D7D-8AA4-7999E1945850.1] [core] Unrecognized response from outbound layer: 903 : Queueing failed
~~~

This is because the `queue` directory is owned by `root`, but `user=<...>` has been set in `stmpd.ini`. Doing `chown <UID> queue` allows queueing to succeed.

### Steps to reproduce

Setup an outbound server using the [guide](https://haraka.github.io/manual/tutorials/SettingUpOutbound.html), but add `user=<...>` and `group=<...>` to `stmp.ini`. It seems haraka creates the `queue` directory automatically, but doesn't set correct permissions/ownership on it?

--- a/Changes.md
+++ b/Changes.md
@@ -4,6 +4,7 @@
 #### Fixed
 
 - fix(outbound): allow LHLO over insecure socket if TLS is forced #3278
+- fix(outbound): chown queue dir after creation #3291
 
 ### [3.0.3] - 2024-02-07
 

--- a/outbound/queue.js
+++ b/outbound/queue.js
@@ -1,10 +1,11 @@
 'use strict';
 
-const async       = require('async');
+const child_process = require('child_process');
 const fs          = require('fs');
 const path        = require('path');
 
-const { Address }     = require('address-rfc2821');
+const async       = require('async');
+const { Address } = require('address-rfc2821');
 const config      = require('haraka-config');
 
 const logger      = require('../logger');
@@ -270,6 +271,17 @@ exports.ensure_queue_dir = () => {
     logger.logdebug(`[outbound] Creating queue directory ${queue_dir}`);
     try {
         fs.mkdirSync(queue_dir, 493); // 493 == 0755
+        const cfg = config.get('smtp.ini');
+        let uid
+        let gid
+        if (cfg.user) uid = child_process.execSync(`id -u ${cfg.user}`).toString().trim();
+        if (cfg.group) gid = child_process.execSync(`id -g ${cfg.group}`).toString().trim();
+        if (uid && gid) {
+            fs.chown(queue_dir, uid, gid)
+        }
+        else if (uid) {
+            fs.chown(queue_dir, uid)
+        }
     }
     catch (err) {
         if (err.code !== 'EEXIST') {

